Alternate font names processing added, external config moved into jar

Intelligent2013 opened the pull request.

ronaldtse commented:
@Intelligent2013 Thanks for this! I actually wonder whether we should extract all the font triplet information into the YAML file, in this case we won't need to worry about pdf_font_config anymore?

Intelligent2013 commented:
@ronaldtse 
> I actually wonder whether we should extract all the font triplet information into the YAML file, in this case we won't need to worry about pdf_font_config anymore

If we'll use YAML file for font description, we need to parse it and convert it again in pdf_font_config format.

I've updated pdf_font_config and we can specify font file name without full path. mn2pdf will find specified font in user's font folder (passed as 1st param in cmd), then in then system font folder.
So don't need to worry about font paths. You can just put a non-system font files into some folder (or sub-folders), and mn2pdf will find it by file name (or alternate file name).
Also, I've added a 'alternate' tag into pdf_font_config to specify an alternate font file name like this:
<alternate embed-url="Cambria Bold.ttf"/>
So don't need to use another fontmap.yml file.

Sure, we can use some YAML file similar pdf_font_config, but in any case we have to convert it into fop config xml file, or parsed into API FOP method calls.

ronaldtse commented:
@Intelligent2013 I tried merging the conflicts but am uncomfortable doing so... can you help resolve them? Thanks!
